File: Year-2022/LeetCode-November-challenge/RemoveAllAdjacentDuplicatesInString.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    string removeDuplicates(string s) {
        string res;
        for(int i = 0; i < s.size(); ++i){
            res.push_back(s[i]);
            while(!res.empty() && res.back() == s[i + 1]){
                i++;
                res.pop_back();
            }
        }
        return res;
    }
};
int main(){
    string s = "abbaca";
    Solution *S = new Solution();
    cout<<S->removeDuplicates(s)<<endl;
    return 0;
}

File: Year-2022/LeetCode-November-challenge/RemoveAllAdjacentDuplicatesInString.js
var removeDuplicates = function (s) {
    let chars = s.split('');
    let res = [];
    for (let x of chars) {
        if (res.length && res[res.length - 1] == x) {
            res.pop();
        } else res.push(x);
    }
    return res.join('');
};
let s = "abbaca";
console.log(removeDuplicates(s));
